src/management/ManagementTokenProvider.js

```diff
@@ -0,0 +1,137 @@
+var ArgumentError = require('rest-facade').ArgumentError;
+var assign = Object.assign || require('object.assign');
+var AuthenticationClient = require('../auth');
+var memoizer = require('lru-memoizer');
+var Promise = require('bluebird');
+
+var DEFAULT_OPTIONS = { enableCache: true };
+
+/**
+ * @class ManagementTokenProvider
+ * Auth0 Management API Token Provider.
+ * @constructor
+ * @memberOf module:management
+ *
+ * @param {Object}  options                         Options for the ManagementTokenProvider.
+ * @param {String}  options.domain                  ManagementClient server domain.
+ * @param {String}  options.clientId                Non Interactive Client Id.
+ * @param {String}  options.clientSecret            Non Interactive Client Secret.
+ * @param {String}  options.audience                Audience of the Management API.
+ * @param {String}  options.scope                   Non Interactive Client Scope.
+ * @param {Boolean} [options.enableCache=true]      Enabled or Disable Cache
+ * @param {Number}  [options.cacheTTLInSeconds]     By default the `expires_in` value will be used to determine the cached time of the token, this can be overridden.
+ */
+var ManagementTokenProvider = function (options) {
+  if (!options || typeof options !== 'object') {
+    throw new ArgumentError('Options must be an object');
+  }
+
+  var params = assign({}, DEFAULT_OPTIONS, options);
+  
+  if (!params.domain || params.domain.length === 0) {
+    throw new ArgumentError('Must provide a domain');
+  }
+
+  if (!params.clientId || params.clientId.length === 0) {
+    throw new ArgumentError('Must provide a clientId');
+  }
+
+  if (!params.clientSecret || params.clientSecret.length === 0) {
+    throw new ArgumentError('Must provide a clientSecret');
+  }
+
+  if (!params.audience || params.audience.length === 0) {
```

Comment:
The audience in the client constructor is optional and it should be since it can be build from the domain + `/api/v2/` unless its provided.

Also I see that the scope is optional in the constructor but not here so we need to be consistent in the docs.

Reply:
maybe consider having a generic TokenProvider that requires an audience value and making ManagementTokenProvider a sub-class that explicitly sets "audience", in it's constructor options, to `"https://" + domain + "/api/v2"` before passing them on to the base class constructor.

P.S. the reason I think it is important to have/provide a generic TokenProvider is because that is what you want if one happens to be interacting with an Extension API (e.g. Auth0's official Authorization Extension) ... having & exposing a generic TokenProvider means users of extension APIs have a first class helper/tool for managing `access_token`s